components/authn-service/server/server.go

```diff
@@ -193,16 +194,26 @@ func newServer(ctx context.Context, c Config) (*Server, error) {
 			tokenID = existingID
 		}
 
-		_, err := policiesClient.AddPolicyMembers(ctx, &authz.AddPolicyMembersReq{
-			Id:      IngestPolicyID,
-			Members: []string{fmt.Sprintf("token:%s", tokenID)},
-		})
-		if err != nil {
-			s.logger.Warn(errors.Wrap(err, "there was an error granting the legacy data collector token ingest access").Error())
-			s.logger.Warn(fmt.Sprintf("please manually add token with ID %q to the policy with ID %q", tokenID, IngestPolicyID))
-		}
-	}
-
+		/*
+			Below code is only execute when you add token in config at the time of deployment
+			Adding the sleep, is just the tempory solution, this might work
+		*/
+		for i := 4; i > 0; i-- {
+			_, err := policiesClient.AddPolicyMembers(ctx, &authz.AddPolicyMembersReq{
+				Id:      IngestPolicyID,
+				Members: []string{fmt.Sprintf("token:%s", tokenID)},
+			})
+			if err != nil {
+				s.logger.Warn(errors.Wrap(err, "there was an error granting the legacy data collector token ingest access").Error())
+				s.logger.Warn(fmt.Sprintf("please manually add token with ID %q to the policy with ID %q", tokenID, IngestPolicyID))
+			} else {
+				s.logger.Warn("exiting for loop")
+				break
+			}
+			s.logger.Warn(fmt.Sprintf(" Adding the sleep for %d seconds ", i))
+			time.Sleep(time.Duration(i) * time.Second)
```

Comment:
Having a variable sleep time may not be a good idea. The last iteration will have 1 second delay.

Reply:
Have we tested that this much delay is enough always?

Reply:
yes..i have tested that.